src/cppmangle.d

```diff
@@ -893,6 +893,13 @@ static if (TARGET_LINUX || TARGET_OSX || TARGET_FREEBSD || TARGET_OPENBSD || TAR
                 store(null);
             store(t);
         }
+
+        final const(char)* mangle_typeinfo(Dsymbol s)
+        {
+            buf.writestring("_ZTI");
```

Comment:
do_catch is not defined to work properly unless you pass it a valid C++ type_info. The trouble comes when you pass it a D TypeInfo.